subprojects/core/src/main/java/org/gradle/api/internal/changedetection/changes/NoHistoryArtifactState.java

```diff
@@ -80,4 +82,9 @@ public FileCollection getOutputFiles() {
     public OverlappingOutputs getOverlappingOutputs() {
         return null;
     }
+
+    @Override
+    public Set<File> getPreviousOutputs() {
+        return null;
+    }
```

Comment:
I guess there's no way to avoid using `null` here, right? I'm not saying we should start right now, but using `Optional` to distinguish between a non-existent and an empty collection would be nicer.